homeassistant/components/imeon_inverter/config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Config flow for Imeon integration."""
+
+import logging
+from typing import Any
+
+from imeon_inverter_api.inverter import Inverter
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ADDRESS, CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ImeonInverterConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the initial setup flow for Imeon Inverters."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> config_entries.ConfigFlowResult:
+        """Handle the user step for creating a new configuration entry."""
+
+        schema = vol.Schema(
+            {
+                vol.Required(CONF_ADDRESS): str,
+                vol.Required(CONF_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+            }
+        )
+
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            connection = False
+
+            async with Inverter(user_input[CONF_ADDRESS]) as client:
+                try:
+                    # Check connection
+                    connection = await client.login(
+                        user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+                    )
+
+                except TimeoutError:
+                    errors["base"] = "cannot_connect"
+
+                except ValueError as e:
+                    if "Host invalid" in str(e):
+                        errors["base"] = "invalid_host"
+
+                    elif "Route invalid" in str(e):
+                        errors["base"] = "invalid_route"
+
+                    else:
+                        errors["base"] = "unknown"
+
+                else:
+                    if connection is True:
```

Comment:
```suggestion
if not errors:
```
the you don't need the connection var and we save one indent

Reply:
connection var is used to check if the credentials are valid, there may be no errors but the connection var could be false

Reply:
ah that'll actually returns false
```suggestion
                    if connection:
```

Reply:
no I meant to keep the else, just my bad